packages/launcher/src/renderer/launcher/contacts/InviteContactModal.js

```diff
@@ -0,0 +1,434 @@
+//@flow
+import React, { Component } from 'react'
+import styled from 'styled-components/native'
+import { Text } from '@morpheus-ui/core'
+import { graphql, createFragmentContainer } from 'react-relay'
+import Loader from '../../UIComponents/Loader'
+
+import rpc from '../rpc'
+import FormModalView from '../../UIComponents/FormModalView'
+import { EnvironmentContext } from '../RelayEnvironment'
+import applyContext, { type ContextProps } from '../LauncherContext'
+
+import type { InviteContactModal_contact as Contact } from './__generated__/InviteContactModal_contact.graphql.js'
+
+export type TransactionType = 'invite' | 'retrieveStake' | 'declineInvite'
+
+type Props = ContextProps & {
+  closeModal: () => void,
+  contact: Contact,
+  type: TransactionType,
+}
+
+type TXParams = {
+  gasPriceGwei: string,
+  maxCost: string,
+  stakeAmount: string,
+}
+
+type State = {
+  txProcessing: boolean,
+  error?: ?string,
+  txParams?: ?TXParams,
+  declinedTXHash?: ?string,
+  withdrawTXHash?: ?string,
+  balances?: {
+    eth: string,
+    mft: string,
+  },
+  invitePending?: {
+    error?: string,
+    state:
+      | 'awaiting_approval'
+      | 'sending_approval'
+      | 'approved'
+      | 'sending_invite'
+      | 'invite_sent'
+      | 'error',
+  },
+}
+
+const MFT_TOKEN_ADDRESSES = {
```

Comment:
Could you move this to the `constants.js` file please? We should have a single reference to these addresses rather than having to copy them in component modules.